ilastik/widgets/hdf5SubvolumeSelectionDialog.py

```diff
@@ -18,50 +18,78 @@
 # on the ilastik web site at:
 #          http://ilastik.org/license.html
 ###############################################################################
+
+from typing import List, Optional
+
+from PyQt5.QtCore import Qt
 from PyQt5.QtWidgets import (
-    QDialogButtonBox,
     QButtonGroup,
     QComboBox,
     QDialog,
+    QDialogButtonBox,
     QLabel,
     QLineEdit,
+    QListWidget,
     QTextEdit,
     QVBoxLayout,
     QWidget,
 )
 
-from PyQt5.QtCore import Qt
-
 from lazyflow.utility import globList
 
 
 class SubvolumeSelectionDlg(QDialog):
-    """
-    A window to ask the user to choose between multiple HDF5 datasets in a single file.
+    """A window to ask the user to choose between multiple HDF5 datasets in a single file.
+
+    If multi is True, user can select multiple datasets at once.
     """
 
-    def __init__(self, datasetNames, parent):
+    def __init__(self, datasetNames, parent, *, multi=False):
         super().__init__(parent)
-        label = QLabel(
-            "Your HDF5/N5 File contains multiple image volumes.\n" "Please select the one you would like to open."
-        )
-
-        self.combo = QComboBox()
-        for name in datasetNames:
-            self.combo.addItem(name)
-
-        buttonbox = QDialogButtonBox(Qt.Horizontal, parent=self)
-        buttonbox.setStandardButtons(QDialogButtonBox.Ok | QDialogButtonBox.Cancel)
-        buttonbox.accepted.connect(self.accept)
-        buttonbox.rejected.connect(self.reject)
+        self.multi = multi
 
         layout = QVBoxLayout()
-        layout.addWidget(label)
-        layout.addWidget(self.combo)
-        layout.addWidget(buttonbox)
-
         self.setLayout(layout)
 
+        label_lines = ["<p>Your file contains multiple images.</p>"]
+        if multi:
+            label_lines.append("<p>Select images that you would like to open.</p>")
+        else:
+            label_lines.append("<p>Select image that you would like to open.</p>")
+        layout.addWidget(QLabel("".join(label_lines), parent=self))
+
+        if multi:
+            self.items = QListWidget(parent=self)
+            self.items.setSelectionMode(QListWidget.MultiSelection)
```

Comment:
Is there any way we can get `Shift + Click` working on this? Otherwise people have to click every single one of the datasets they're trying to add =)

Reply:
All the "normal" keyboard shortcuts should work, including Ctrl-A. Do you think we should mention them in a help text?

Reply:
Now that you mention that, maybe it's the easiest (and most helpful) to add this information in a help-text, as you suggest. I would probably always try the standard things anyway, but I'm sure users will appreciate this info!